src/app/api/graphql/route.ts

```diff
@@ -38,9 +38,13 @@ const resolvers = {
             await mongooseConnect();
             return CommentModel.find({ event_id }).sort({ createdAt: -1 }).populate("author");
         },
-        locations: async (parent: any, { name }: { name: String }) => {
+        locations: async (parent: any) => {
             await mongooseConnect();
-            return LocationModel.find({ name });
+            return LocationModel.find({ name }).sort({ createdAt: -1 });
```

Comment:
You need to sort not on a date of creation, but on a relevance.  Also, it will make sense to accept name or coordinates or other location.